typescript/infra/scripts/validators/print-latest-checkpoints.ts

```diff
@@ -0,0 +1,124 @@
+import { ValidatorAnnounce__factory } from '@hyperlane-xyz/core';
+import {
+  ChainMap,
+  defaultMultisigConfigs,
+  getValidatorFromStorageLocation,
+} from '@hyperlane-xyz/sdk';
+import {
+  Address,
+  LogFormat,
+  LogLevel,
+  configureRootLogger,
+  rootLogger,
+} from '@hyperlane-xyz/utils';
+
+import { isEthereumProtocolChain } from '../../src/utils/utils.js';
+import { getArgs, withChainsRequired } from '../agent-utils.js';
+import { getHyperlaneCore } from '../core-utils.js';
+
+function getHttpsUrl(storageLocation: string): string {
+  if (storageLocation.startsWith('s3://')) {
+    // Convert s3:///bucket-name/region to https://bucket-name.s3.region.amazonaws.com
+    const [, , bucket, region] = storageLocation.split('/');
+    return `https://${bucket}.s3.${region}.amazonaws.com`;
+  } else if (storageLocation.startsWith('gs://')) {
+    // Convert gs://bucket-name to https://storage.googleapis.com/bucket-name
+    const bucket = storageLocation.replace('gs://', '');
+    return `https://storage.googleapis.com/${bucket}`;
+  }
```

Comment:
iirc we do already have this logic somewhere in the TS codebase, at least for aws

Reply:
indeed, this function is a relic before i switched to calling `getLatestCheckpointUrl`